src/features/governance/hooks/useProposalApprovers.ts

```diff
@@ -0,0 +1,121 @@
+import { governanceABI, multiSigABI } from '@celo/abis-12';
+import { useMemo } from 'react';
+import { StaleTime } from 'src/config/consts';
+import { Addresses } from 'src/config/contracts';
+import { useProposalDequeueIndex } from 'src/features/governance/hooks/useProposalQueue';
+import { encodeFunctionData } from 'viem';
+import { useReadContract, useReadContracts } from 'wagmi';
+
+// @returns addresses of approvers that have confirmed the proposal
+export function useProposalApprovers(proposalId: number) {
```

Comment:
maybe a bit nitpicky, but can we refactor this into smaller hooks? Not that it matters much but it _feels_ like it's reading a lot of different contracts that can maybe be their own hook? Not blocking though.